src/zql/util/iterables.ts

```diff
@@ -8,6 +8,12 @@ export function* mapIter<T, U>(
   }
 }
 
+/**
+ * Flat maps the items returned from the iterable.
+ *
+ * `iter` is a lambda that returns an iterable
+ * so this function can return an `IterableIterator`
```

Comment:
Maybe it should take an IterableIterator then?

Also, These should be compatible with the standard methods to simplify transition to them later:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/flatMap

Reply:
going to iterate on this separately. `IterableIterator` is also getting exhausted on me :/

E.g.,

```
x = flatMapIter(iter, fn);
console.log([...x])
console.log([...x])
```

second spread is empty :/

Or maybe a better question to answer for myself is why would we need to pull the iterable twice. Seems like it should only ever happen once.

Reply:
To get the second log to be non empty `x` needs to be something that has `[Symbol.iterator](): Iterator` which returns a new iterator every time.

This API was something that I felt we didn't get quite right for ES6. It is always confusing.